### PR TITLE
Fix homestead [WIP]

### PR DESCRIPTION
**Pull request by cdetrio**
The first commit prevents `runState.gasLimit` from going negative and failing the test case [call_OOG_additionalGasCosts1](https://github.com/ethereum/tests/blob/e8ba92a9e20a252d215cbc95d314eba22b1d79c5/src/GeneralStateTestsFiller/stCallCodes/call_OOG_additionalGasCosts1Filler.json) under Homestead.

The second commit fixes the [shallowStack_d10g0v0_Homestead](https://github.com/ethereum/tests/blob/cffb263d6c9075e1396e72d5d8febc21329b6600/src/GeneralStateTestsFiller/stStackTests/shallowStackFiller.json#L42) test case, which executes the bytecode `0x60010b600055`:

```
[1] PUSH1 0x01 
[2] SIGNEXTEND 
[4] PUSH1 0x00 
[5] SSTORE 
```
and should throw an OOG due to a stack underflow.

The third commit fixes some failing edge cases in the [stackOverflow](https://github.com/ethereum/tests/blob/cffb263d6c9075e1396e72d5d8febc21329b6600/src/GeneralStateTestsFiller/stStackTests/stackOverflowFiller.json) tests.

### CODE REVIEW

File: lib/opFns.js [hunk 1]
```diff
@@ -555,8 +555,12 @@ module.exports = {
         return
       }
       if (!exists) {
-        // can't wrap because we are in a callback
-        runState.gasLeft.isub(new BN(fees.callNewAccountGas.v))
+        try {
+          subGas(runState, new BN(fees.callNewAccountGas.v))
+        } catch (e) {
+          done(e.error)
```
Comment (axic) [inline lib/opFns.js]: Shouldn't this be within a `if (homestead)` check?

File: lib/opcodes.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ const codes = {
   0x08: ['ADDMOD', 8, 3, 1, false],
   0x09: ['MULMOD', 8, 3, 1, false],
   0x0a: ['EXP', 10, 2, 1, false],
-  0x0b: ['SIGNEXTEND', 5, 1, 1, false],
+  0x0b: ['SIGNEXTEND', 5, 2, 1, false],
```
Comment (axic) [inline lib/opcodes.js]: Can you review the rest of the opcodes? Others might have bugs :)

File: lib/runCode.js [hunk 3]
```diff
@@ -131,6 +128,10 @@ module.exports = function (opts, cb) {
       return done(ERROR.STACK_UNDERFLOW)
     }
 
+    if (runState.stack.length - opInfo.in + opInfo.out > 1024) {
```
Comment (axic) [inline lib/runCode.js]: Can you enclose the left side in parentheses?

**Comment by axic** [comment]
Note that #84 will also be needed to complete homestead.

File: lib/opFns.js (hunk 1 above)
Comment (cdetrio) [inline lib/opFns.js]: I think the rule that a 0-value call creates a new account (and pays the new account gas fee) was there in frontier.

Comment (axic) [inline lib/opFns.js]: Right, this commit log was a bit misleading. It fixes a bug when there wasn't sufficient gas left.

**Comment by axic** [comment]
There is another bug, `CALLCODE` doesn't check for gas underflow for the stipend. Search for `isub`.
